keras/src/visualization/plot_image_gallery.py

```diff
@@ -0,0 +1,165 @@
+import math
+
+import numpy as np
+
+from keras.src import backend
+from keras.src import ops
+from keras.src.api_export import keras_export
+from keras.src.layers.preprocessing.image_preprocessing.base_image_preprocessing_layer import (  # noqa: E501
+    BaseImagePreprocessingLayer,
+)
+
+try:
+    import matplotlib.pyplot as plt
+except ImportError:
+    plt = None
+
+
+def _extract_image_batch(images, num_images, batch_size):
+    """Extracts a batch of images for plotting.
+
+    Args:
+        images: The 4D tensor or NumPy array of images.
+        num_images: The number of images to extract.
+        batch_size: The original batch size of the images.
+
+    Returns:
+        A 4D tensor or NumPy array containing the extracted images.
+
+    Raises:
+        ValueError: If `images` is not a 4D tensor/array.
+    """
+
+    if len(ops.shape(images)) != 4:
+        raise ValueError(
+            "`plot_images_gallery()` requires you to "
+            "batch your `np.array` samples together."
+        )
+    num_samples = min(num_images, batch_size)
+    sample = images[:num_samples, ...]
+
+    return sample
+
+
+@keras_export("keras.visualization.plot_image_gallery")
+def plot_image_gallery(
+    images,
+    value_range,
+    rows=None,
+    cols=None,
+    scale=2,
+    path=None,
+    show=None,
+    transparent=True,
+    dpi=60,
+    legend_handles=None,
+    data_format=None,
+):
+    """Displays a gallery of images.
+
+    Args:
+        images: A 4D tensor or NumPy array of images. Shape should be
+           `(batch_size, height, width, channels)`.
+        value_range: A tuple specifying the value range of the images
```

Comment:
Require this to be specified is a bit user-unfriendly. I would recommend defaulting to (0, 255), which is what image plotting utilities normally expect.

Reply:
Yup make sense. Done!